### PR TITLE
Rule cleanup

### PR DESCRIPTION
**Pull request by davidwengier**
From #3815:
- [x] Split `File` and `BrowseObject`. 
-- Done for Compile, Content, EmbeddedResource, None and AdditionalFiles
- [x] Remove all localizable strings from rules that are not in the `BrowseObject` context. These strings do not show.
-- Removed all display names and descriptions
- [x] Remove empty xlf files
-- Removed, and prevented generation on build
- [x] Apply PropertyPagesHidden="true" to all `BrowseObject`
-- Was already done
- [x] Remove Identity/FileNameAndExtension/URL/Extension from `File`
-- Removed

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/None.xaml [hunk 1]
```diff
@@ -2,99 +2,40 @@
...
-  </Rule.DataSource>
-  <Rule.Categories>
-    <Category Name="Advanced" DisplayName="Advanced" />
-    <Category Name="Misc" DisplayName="Misc" />
-  </Rule.Categories>
+    <Rule.DataSource>
+        <DataSource Persistence="ProjectFile" HasConfigurationCondition="False" ItemType="None" SourceOfDefaultValue="AfterContext" />
+    </Rule.DataSource>
+    <Rule.Categories>
+        <Category Name="Advanced" />
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/None.xaml]: Categories can also be removed from the "file" context. We no longer/never show the results of this file in any UI.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/xlf/None.xaml.pt-BR.xlf [hunk 2]
```diff
@@ -1,127 +1,6 @@
...
-        <target state="translated">Um valor indicando se esse é um arquivo gerado.</target>
-        <note />
-      </trans-unit>
-      <trans-unit id="StringProperty|CustomTool|Description">
-        <source>DTE Property for accessing the Generator property.</source>
-        <target state="translated">Propriedade DTE para acessar a propriedade Generator.</target>
-        <note />
-      </trans-unit>
-    </body>
+    <body />
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/xlf/None.xaml.pt-BR.xlf]: I think we can now opt these out of localization - but I'm not sure how.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/xlf/None.xaml.pt-BR.xlf]: Which will also involve removing the files.

Comment (davidwengier) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/xlf/None.xaml.pt-BR.xlf]: The file is empty, so certainly nobody will bother localizing anything in them, but not having localization files would be nicer. Will add a task for it

**Comment by Pilchie** [comment]
Very happy to see progress on this finally, after I haven't had time in forever...

**Comment by davidwengier** [comment]
This is ready for review, for the splitting up of rules. I logged #4104 and #4105 for the remaining items, but there is no reason to hold this up.

Any pointers for any extra testing needed for this would be appreciated.

**Comment by davidwengier** [comment]
Attn @dotnet/project-system I'm gonna merge this tomorrow so if you want to review it, speak now of forever hold your peace :)

**Pull request merged**